sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -337,11 +337,11 @@ class DataFrame protected[sql](
    * {{{
    *   // Scala:
    *   import org.apache.spark.sql.functions._
-   *   df1.join(df2, "outer", $"df1Key" === $"df2Key")
+   *   df1.join(df2, $"df1Key" === $"df2Key", "outer")
    *
    *   // Java:
    *   import static org.apache.spark.sql.functions.*;
-   *   df1.join(df2, "outer", col("df1Key") === col("df2Key"));
+   *   df1.join(df2, col("df1Key") === col("df2Key"), "outer");
```

Comment:
actually while you are at it, do you mind fixing the java example? it definitely wouldn't work with col("df1key") === col("df2key").

I think you need col("df1Key").equalTo(col("df2Key"))